DataFormats/Math/test/testAtan2.cpp

```diff
@@ -26,14 +26,11 @@ inline T deltaPhi (T phi1, T phi2) {
 inline bool phiLess(float x, float y) {
   auto ix = phi2int(toPhi(x));
   auto iy = phi2int(toPhi(y));
-
-  return (ix-iy)<0;
+  return ix<iy; // avoid overflows on signed ints//(ix-iy)<0;
 
```

Comment:
of course now the result is wrong half of the times

Reply:
Just change resulting type from `int` (32-bit) to `long long int` (64-bit) to contain the result of `(ix-iy)`.

Reply:
Ah, this was intentional. In C/C++ this is undefined and basically depends on hardware, but compiler can take advantage of this while doing optimizations. IIUC from DavidL, this what happens with Clang. I suggested to try `-fno-strict-overflow`  and `-fwrapv`, not sure if those are supported by Clang.

Reply:
That's not the intended behavior of this function, it intentionally wrap to define "<" on a circle
(well a sort of less, a<b<c does not guarantee a<c)